src/components/multiSelect/MultiSelectBottomSheet.tsx

```diff
@@ -0,0 +1,271 @@
+import GorhomBottomSheet, { BottomSheetBackdrop, BottomSheetProps } from '@gorhom/bottom-sheet'
+import { BottomSheetDefaultBackdropProps } from '@gorhom/bottom-sheet/lib/typescript/components/bottomSheetBackdrop/types'
+import React, { useCallback, useMemo, useRef } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Keyboard, StyleSheet, Text, View } from 'react-native'
+import FastImage from 'react-native-fast-image'
+import { ScrollView } from 'react-native-gesture-handler'
+import { useSafeAreaFrame, useSafeAreaInsets } from 'react-native-safe-area-context'
+import BottomSheetScrollView from 'src/components/BottomSheetScrollView'
+import Touchable from 'src/components/Touchable'
+import Checkmark from 'src/icons/Checkmark'
+import Colors from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+
+const ITEM_HEIGHT = 60
+const MAX_ITEMS_IN_VIEW = 5
+
+interface MultiSelectBottomSheetProps<T extends string> {
+  forwardedRef: React.RefObject<GorhomBottomSheet>
+  onChange?: BottomSheetProps['onChange']
+  onClose?: () => void
+  onOpen?: () => void
+  handleComponent?: BottomSheetProps['handleComponent']
+  selectedItems: Record<T, boolean>
```

Comment:
my initial thought was this was a list of selected items, and I was searching for a prop containing all items before realizing this includes all items with a true / false. I was looking at some other packages to see what interface they support (e.g., https://www.npmjs.com/package/react-multi-select-component), could we adopt something similar? I guess you can also infer all items from `textAndIconMap` since it should include all keys for rendering the item, which can act analogous to the options param in the package. I think you can also avoid the `as T[]` casting if you change the shape of `textAndIconMap` slightly. Or is there a strong reason to have consumers pass in an object for selectedItems instead of an array?

Reply:
An object just made more sense to me when I was initially thinking about it. Static options, O(1) for updates to the object since we always know the keys. The difference between that an and o(n) array.filter to update an array of items is really unimportant though at this scale.

I can update to an array similar to the props that you linked!